### PR TITLE
17 add overflow crop method

### PR DESCRIPTION
**Pull request by chandley5641**
Reliant on [https://github.com/tonkintaylor/rastr/pull/66](url). Wait for this to be reviewed first. 

Adding the `overflow` method to `Raster.crop` such that cells overlapping a bounding box are not clipped.

### CODE REVIEW

File: src/rastr/raster.py [hunk 1]
```diff
@@ -629,15 +629,16 @@ def extrapolate(self, method: Literal["nearest"] = "nearest") -> Self:
...
+        strategy: Literal["underflow", "overflow"] = "underflow",
     ) -> Self:
         """Crop the raster to the specified bounds.
 
         Args:
             bounds: A tuple of (minx, miny, maxx, maxy) defining the bounds to crop to.
             strategy:   The cropping strategy to use. 'underflow' will crop the raster
                         to be fully within the bounds, ignoring any cells that are
+                        partially outside the bounds. 'overflow' will crop the raster
+                        to include all cells that intersect the bounds, even if they are
```
Comment (nathanjmcdougall) [inline src/rastr/raster.py:641]: I think let's reword this a bit, something like
```
'overflow' will instead include cells that intersect the bounds, ensuring the bounds area remains covered with cells.
```

File: tests/rastr/test_raster.py [hunk 2]
```diff
@@ -775,6 +775,18 @@ def test_border_cells_cropped(self, base_raster: RasterModel):
         assert cropped.raster_meta.crs == base_raster.raster_meta.crs
         assert cropped.raster_meta.transform == expected_transform
 
+    def test_overflow_doesnt_crop(self, base_raster: RasterModel):
```
Comment (nathanjmcdougall) [inline tests/rastr/test_raster.py:778]: I think we need another test which tests behaviour where an actual crop occurs. This test is good but is only considers the idempotent case where the raster is already cropped

**Review by nathanjmcdougall — CHANGES_REQUESTED** [review]

File: tests/rastr/test_raster.py (hunk 2 above)
Comment (chandley5641) [reply tests/rastr/test_raster.py:778]: I've added this. I also modified it so that when the boundary aligns with cell boundaries it crops to match the bbox perfectly (with a boundary case test).

**Review by nathanjmcdougall — APPROVED** [review]
